File: test/kotlin/gradle/http-service/success-deployment-smoke/src/main/kotlin/acceptance/http/Router.kt
package acceptance.http
import acceptance.Configuration
import acceptance.service.SummaryService
import java.net.URLDecoder
import java.nio.charset.StandardCharsets
data class Response(val status: Int, val type: String, val body: String)
class Router(private val configuration: Configuration) {
    private val service = SummaryService()
    fun route(target: String): Response {
        if (configuration.status == 503) return Response(503, "text/plain; charset=utf-8", configuration.label)
        val path = target.substringBefore('?')
        if (path == "/api/summary" || configuration.mode == "json") {
            try {
                val query = target.substringAfter('?', "")
                val pair = query.split('&').map { it.split('=', limit = 2) }
                    .firstOrNull { URLDecoder.decode(it[0], StandardCharsets.UTF_8) == "values" }
                val raw = if (path == "/api/summary" && pair != null)
                    URLDecoder.decode(pair.getOrElse(1) { "" }, StandardCharsets.UTF_8) else null
                return Response(200, "application/json; charset=utf-8", service.summarize(raw).toJson())
            } catch (error: IllegalArgumentException) { return Response(400, "text/plain; charset=utf-8", "invalid-values") }
        }
        return Response(200, "text/plain; charset=utf-8", configuration.label)
    }
}

File: test/single-language/php/phpcli/http-service/success-json-api/src/Router.php
<?php
declare(strict_types=1);
namespace Fixture;
final class Router {
    public function __construct(private readonly Configuration $configuration) {}
    public function route(string $target): array {
        $status = $this->configuration->status;
        $body = $this->configuration->label;
        $type = 'text/plain; charset=utf-8';
        $path = parse_url($target, PHP_URL_PATH);
        if ($status !== 503 && ($path === '/api/summary' || $this->configuration->mode === 'json')) {
            parse_str(parse_url($target, PHP_URL_QUERY) ?? '', $query);
            try {
                $raw = $path === '/api/summary' ? ($query['values'] ?? null) : null;
                if ($raw !== null && !is_string($raw)) throw new \InvalidArgumentException('invalid-values');
                $body = (new SummaryService())->summarize($raw)->toJson();
                $type = 'application/json; charset=utf-8';
            } catch (\InvalidArgumentException $error) {
                $status = 400; $body = 'invalid-values';
            }
        }
        return ['status' => $status, 'type' => $type, 'body' => $body];
    }
}
